File: client-scalaz/src/main/scala/org/openapitools/client/api/AchievementResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import AchievementResponse._

case class AchievementResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
achievementId: Option[Long],
title: Option[String],
description: Option[String],
analyticsTag: Option[String],
rankType: Option[String],
rankIncrement: Option[Integer],
validate: Option[Boolean],
appName: Option[String],
active: Option[Boolean],
updated: Option[Long],
created: Option[Long],
tierCount: Option[Integer],
minIncrement: Option[Integer],
maxIncrement: Option[Integer],
tiers: Option[List[AchievementTierResponse]],
returning: Option[String])

object AchievementResponse {
  import DateTimeCodecs._

  implicit val AchievementResponseCodecJson: CodecJson[AchievementResponse] = CodecJson.derive[AchievementResponse]
  implicit val AchievementResponseDecoder: EntityDecoder[AchievementResponse] = jsonOf[AchievementResponse]
  implicit val AchievementResponseEncoder: EntityEncoder[AchievementResponse] = jsonEncoderOf[AchievementResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/achievement_response.clj
(ns sirqul-io-t-platform.specs.achievement-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.achievement-tier-response :refer :all]
            )
  (:import (java.io File)))


(def achievement-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :achievementId) int?
   (ds/opt :title) string?
   (ds/opt :description) string?
   (ds/opt :analyticsTag) string?
   (ds/opt :rankType) string?
   (ds/opt :rankIncrement) int?
   (ds/opt :validate) boolean?
   (ds/opt :appName) string?
   (ds/opt :active) boolean?
   (ds/opt :updated) int?
   (ds/opt :created) int?
   (ds/opt :tierCount) int?
   (ds/opt :minIncrement) int?
   (ds/opt :maxIncrement) int?
   (ds/opt :tiers) (s/coll-of achievement-tier-response-spec)
   (ds/opt :returning) string?
   })

(def achievement-response-spec
  (ds/spec
    {:name ::achievement-response
     :spec achievement-response-data}))
